File: programmers/level2/12945.java
//피보나치 수

class Solution {    
  public int solution(int n) {
      int i;
      int[] dp = new int[n+1];
      for(i= 0; i<n+1; i++){
          if(i<2) dp[i]=i;
          else{
              dp[i]=dp[i-1]%1234567+dp[i-2]%1234567;
          }
      }
      return dp[n]%1234567;     
  }
}

File: programmers/level2/12945.js
// 2021.03.13

const solution = (n) => {
    const arr = [0, 1]
    for(let i = 2; i <=n; i++){
        arr.push((arr[i-1] + arr[i-2])%1234567)
    }
    return arr[n]
}
